Update guide to v14 code (#1146)
discordjs/guide: guide/interactions/modals.md

});
```

Restart your bot and invoke the `/ping` command again. You should see a popup form resembling the image below:

<img width=450 src="./images/modal-example.png">

## Receiving modal submissions

Modals are received via an interaction. You can check if a given interaction is a modal by checking if the interaction is a `ModalSubmit`.

```js {1,4}
const { InteractionType } = require('discord.js');

client.on('interactionCreate', interaction => {
	if (interaction.type !== InteractionType.ModalSubmit) return;
	console.log(interaction);
});
```

## Responding to modal submissions

The `ModalSubmitInteraction` class provides the same methods as the `CommandInteraction` class. These methods behave equally:

- `reply()`
- `editReply()`
- `deferReply()`
- `fetchReply()`
- `deleteReply()`
- `followUp()`

> ```js{1,4-6}

Reviewer: Not confident this is right.

client.on('interactionCreate', interaction => {
	if (interaction.type !== InteractionType.ModalSubmit) return;
	if (interaction.customId === 'myModal') {
		await interaction.reply({ content: 'Your submission was recieved successfully!' });
	}
});
```

## Extracting data from modal submissions

You'll most likely need to read the data sent by the user in the modal. You can do this by accessing the `#fields` instance field on the interaction. From there you can call `#getTextInputValue` with the custom id of the text input to get the value.

> ```js

Reviewer: The above text talked about accessing the fields and text input values. The relevant lines should be highlighted here.

## Receiving modal submissions

> Modals are received via an interaction. You can check if a given interaction is a modal by checking if the interaction is a `ModalSubmit`.

Reviewer 2: ```suggestion
Modals are received via an interaction. You can check if a given interaction is a modal by checking if the `Interaction.type` is `ModalSubmit`.
```

```js {1,4}
const { InteractionType } = require('discord.js');

client.on('interactionCreate', interaction => {
	if (interaction.type !== InteractionType.ModalSubmit) return;
	console.log(interaction);
});
```

## Responding to modal submissions

> The `ModalSubmitInteraction` class provides the same methods as the `CommandInteraction` class. These methods behave equally:

Author: [`ModalSubmitInteraction#isFromMessage`](https://discord.js.org/#/docs/discord.js/main/class/ModalSubmitInteraction?scrollTo=isFromMessage) in a tip box?

> The `ModalSubmitInteraction` class provides the same methods as the `CommandInteraction` class. These methods behave equally:

Author: ![image](https://user-images.githubusercontent.com/99907738/180308244-8c22a05c-05ea-4331-b4f8-ab18d0b4c0ab.png)
duno, added something like that. let me know if there's anything to be changed

- `reply()`
- `editReply()`
- `deferReply()`
- `fetchReply()`
- `deleteReply()`
- `followUp()`

> ```js{1,4-6}

Reviewer 2: ```suggestion
```js{1,3-5}
```

> client.on('interactionCreate', interaction => {

Reviewer 3: This is using `await` on L4
```suggestion
client.on('interactionCreate', async interaction => {
```

## Receiving modal submissions

Modals are received via an interaction. You can check if a given interaction is a modal by checking if the <DocsLink path="class/BaseInteraction?scrollTo=type"/> is `ModalSubmit`. 

```js {1,4}
const { InteractionType } = require('discord.js');

client.on('interactionCreate', interaction => {
	if (interaction.type !== InteractionType.ModalSubmit) return;
	console.log(interaction);
});
```

## Responding to modal submissions

The `ModalSubmitInteraction` class provides the same methods as the `CommandInteraction` class. These methods behave equally:

- `reply()`
- `editReply()`
- `update()`
- `deferReply()`
- `deferUpdate()`
- `fetchReply()`
- `deleteReply()`
- `followUp()`

```js{1,3-5}
client.on('interactionCreate', async interaction => {
	if (interaction.type !== InteractionType.ModalSubmit) return;
	if (interaction.customId === 'myModal') {
		await interaction.reply({ content: 'Your submission was recieved successfully!' });
	}
});
```

::: tip

> If you're using typescript, you can use the <DocsLink path="class/ModalSubmitInteraction?scrollTo=isFromMessage"/> typeguard, to check whether the received interaction was from a `MessageComponentInteraction` to avoid errors.

Reviewer 2: ```suggestion
If you're using typescript, you can use the <DocsLink path="class/ModalSubmitInteraction?scrollTo=isFromMessage"/> typeguard, to make sure the received interaction was from a `MessageComponentInteraction`.
```

> Modals are received via an interaction. You can check if a given interaction is a modal by invoking the `#isModalSubmit()` method:

Reviewer 4: this typeguard is being removed, use `interaction.type === InteractionType.ModalSubmit`

```js {1,4}
const { InteractionType } = require('discord.js');

client.on('interactionCreate', interaction => {
	if (interaction.type !== InteractionType.ModalSubmit) return;
	console.log(interaction);
});
```

## Responding to modal submissions

The `ModalSubmitInteraction` class provides the same methods as the `CommandInteraction` class. These methods behave equally:

- `reply()`
- `editReply()`
- `deferReply()`
- `fetchReply()`
- `deleteReply()`
- `followUp()`

```js{1,4-6}
client.on('interactionCreate', interaction => {

> 	if (!interaction.isModalSubmit()) return;

Reviewer 4: ```diff
- if (!interaction.isModalSubmit()) return;
+ if (!interaction.type === InteractionType.ModalSubmit) return;
```

Reviewer 2: `if (interaction.type !== InteractionType.ModalSubmit) return;`

Reviewer 4: whoops that's a typo lol, fixing

Modals are received via an interaction. You can check if a given interaction is a modal by invoking the `#isModalSubmit()` method:

```js {1,4}
const { InteractionType } = require('discord.js');

client.on('interactionCreate', interaction => {

> 	if (!interaction.type === InteractionType.ModalSubmit) return;

Reviewer 2: ```suggestion
	if (interaction.type !== InteractionType.ModalSubmit) return;
```
`!interaction.type` is coerced to a boolean which will never be equal to an enum value

You'll most likely need to read the data sent by the user in the modal. You can do this by accessing the `#fields` instance field on the interaction. From there you can call `#getTextInputValue` with the custom id of the text input to get the value.

```js
client.on('interactionCreate', interaction => {

> 	if (!interaction.type === InteractionType.ModalSubmit) return;

Reviewer 2: ```suggestion
	if (interaction.type !== InteractionType.ModalSubmit) return;
```
